library/Ivoz/Provider/Infrastructure/Persistence/Doctrine/VoicemailRelUserDoctrineRepository.php

```diff
@@ -40,4 +43,31 @@ public function persist(\Ivoz\Core\Domain\Model\EntityInterface $voicemailRelUse
     {
         parent::persist($voicemailRelUser);
     }
+
+    /**
+     * @return int[]
+     */
+    public function getVoicemailIdsByUser(UserInterface $user): array
+    {
+        $qb = $this->createQueryBuilder('self');
+        $expression = $qb->expr();
+
+        $query = $qb
+            ->where(
+                $expression->eq('self.user', $user->getId())
+            )->getQuery();
+
+        /** @var VoicemailRelUser[] $voicemailRelUsers */
+        $voicemailRelUsers = $query->getResult();
+        return array_map(
+            function (VoicemailRelUserInterface $voicemailRelUser): int {
+                /** @var int $id */
+                $id = $voicemailRelUser
+                    ->getVoicemail()
+                    ->getId();
+                return $id;
+            },
+            $voicemailRelUsers
+        );
```

Comment:
You can make this easier
```suggestion
        $result = $qb
            ->getQuery()
            ->getScalarResult();

        return array_map(
            function ($row): int {
                return (int) $row['voicemailId'];
            },
            $result
        );
```